crates/rpc/rpc-types/src/eth/transaction/typed.rs

```diff
@@ -23,42 +23,45 @@ pub enum TypedTransactionRequest {
 }
 
 impl TypedTransactionRequest {
-    /// coverts a typed transaction request into a primitive transaction
-    pub fn into_transaction(self) -> Transaction {
-        match self {
+    /// Converts a typed transaction request into a primitive transaction.
+    ///
+    /// Returns `None` if any of the following are true:
+    /// - `nonce` is greater than [`u64::MAX`]
```

Comment:
just checked geth reference and nonce on rpc methods is U64, 

I think we can also change this in our types?

Reply:
https://github.com/ethereum/go-ethereum/blob/5b9cbe30f8ca2487c8991e50e9c939d5e6ec3cc2/internal/ethapi/transaction_args.go#L45-L45

Reply:
Is there a reason we're using ruint for these anyway? I don't see any serde or whatever

Reply:
there are two steps here

`TransactionRequest` -> `TypedTransactionRequest` -> `Transaction`

arguably this could also be one step, but having a typed request is still useful